src/app/shared/object-list/search-result-list-element/search-result-list-element.component.ts

```diff
@@ -47,6 +47,25 @@ export class SearchResultListElementComponent<T extends SearchResult<K>, K exten
     return Metadata.allValues([this.object.hitHighlights, this.dso.metadata], keyOrKeys);
   }
 
+  /**
+   * Gets all matching metadata string values from hitHighlights or dso metadata.
+   *
+   * @param {string|string[]} keyOrKeys The metadata key(s) in scope. Wildcards are supported; see [[Metadata]].
+   * @returns {string[]} the matching string values or an empty array.
+   */
+  allMetadataNoExcludingValues(keyOrKeys: string | string[]): string[] {
```

Comment:
It'd be better to move this code into the `allMetadataValues()` method that already exists.  Currently, in this PR, you've removed all usages of `allMetadataValues()`, so that code is no longer used.   So, the simple solution here is to replace the code in `allMetadataValues()` with your new code.

Reply:
Hi @tdonohue Thanks, I agree with you, but at the start I didn't check if this method was used in others parts. I updated the PR with your recommendation.